File: BOJ/problems/boj9298.cpp
#include <iostream>
using namespace std;

int main(int argc, char const *argv[])
{
	cin.tie(0);
	ios_base::sync_with_stdio(false);

	int T, N;
	double ri, ci, rma, cma, rmi, cmi;

	cout << fixed;
	cout.precision(9);

	cin >> T;
	for (int tc=1; tc<=T; ++tc) {
		cin >> N;
		cma = rma = -1001.0;
		cmi = rmi = 1001;
		for (int i=0; i<N; ++i) {
			cin >> ri >> ci;
			rma = rma < ri ? ri : rma;
			cma = cma < ci ? ci : cma;
			rmi = rmi < ri ? rmi : ri;
			cmi = cmi < ci ? cmi : ci;
		}

		double area = (rma - rmi) * (cma - cmi);
		double peri = (rma - rmi + cma - cmi) * 2;
		cout << "Case " << tc << ": Area " << area << ", Perimeter " << peri << '\n'; 
	}
	return 0;
}

File: BOJ/9000-9999/boj9298.py
T = int(input())
for tc in range(T):
	N = int(input())
	rs = cs = 10000
	rb = cb = -10000
	for _ in range(N):
		ri, ci = [float(a) for a in input().split()]
		if ri > rb:
			rb = ri
		if ri < rs:
			rs = ri
		if ci > cb:
			cb = ci
		if ci < cs:
			cs = ci

	area = (rb - rs) * (cb - cs)
	peri = (rb + cb - rs - cs) * 2

	print('Case ', (tc+1), ': Area ', area, ', Perimeter ', peri)
